test/ranking/ranking.test.ts

```diff
@@ -30,4 +34,56 @@ describe('ranking', () => {
       assert.deepEqual(group.items, []);
     });
   });
+
+  describe('comparatorFactory', () => {
+    it('should create a comparator that returns a score difference when passed an orderBy array', () => {
+      const specQ: SpecQuery = {
+        mark: SHORT_ENUM_SPEC,
+        encodings: [
+          {
+            channel: SHORT_ENUM_SPEC,
+            bin: SHORT_ENUM_SPEC,
+            aggregate: SHORT_ENUM_SPEC,
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          },
+          {
+            channel: SHORT_ENUM_SPEC,
+            bin: SHORT_ENUM_SPEC,
+            aggregate: SHORT_ENUM_SPEC,
+            field: 'Q1',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+      const answerSet = generate(specQ, schema, DEFAULT_QUERY_CONFIG);
+      const comparator = comparatorFactory(['aggregationQuality', 'effectiveness'], schema, DEFAULT_QUERY_CONFIG);
+      assert.isNumber(comparator(answerSet[1], answerSet[4]));
```

Comment:
This is a unit test for comparator factory.  It shouldn't call `generate`.  

Readers can't imagine what would be answerSet[1] and [4].  Plus, 1 and 4 are both arbitrary.  
Why don't you follow my suggestion that other day which is just create two SpecQueryModel.